src/HttpClient/HttpClientFactory.php

```diff
@@ -175,4 +128,61 @@ public function create(Options $options): HttpAsyncClientInterface
 
         return new PluginClient($httpClient, $httpClientPlugins);
     }
+
+    /**
+     * @param Options $options
+     *
+     * @return ClientInterface|HttpAsyncClient
```

Comment:
The method return type should be `HttpAsyncClientInterface` as the whole SDK still works using such interface rather than the PSR-18 non-async interface